Initial Phase 3

brycerandolph opened the pull request:
This contains the data and completed constructors for the DataMatrix class as well as the accessors for actualHeight and actualWidth.

apainintheneck committed:
Added the main file Assig4.java with the sample main() from the spec.

apainintheneck committed:
Added BarcodeIO.java.

Contains BarcadIO interface.

apainintheneck committed:
Working on BarcodeImage class.

Finished Methods:
-getPixel
-setPixel

apainintheneck committed:
Added more methods to BarcodeImage class.

Added:
BarcodeImage() -Default constructor
BarcodeImage(String[] strData) -Another Constructor
CheckSize(String[] data) -private method to check for valid strData.

apainintheneck committed:
Added displayToConsole() for testing to BardcodeImage class.

apainintheneck committed:
Added testing for BarcodeImage class.

Testing is in the file testBarcodeImage.java.

apainintheneck committed:
Implemented cloning in BarcodeImage class.

Finished the BarcodeImage.clone() method and added tests in the
testBarcodeImage.java file to show that it is working.

apainintheneck committed:
Added stubs for DataMatrix.java class.

These are the basic stubs. I might have missed one or two though. I
labeled each method or group of methods with each persons name.

brycerandolph committed:
Data, Constructors, and Accessors for the DataMatrix Class

This is the intial commit of the Data, three required constructors, and accessors(actualHeight, actualWidth) for use by the DataMatrix Class.

apainintheneck commented:
There are a few problems that I see here.

1. This branch has merge conflicts that must be resolved before it can be merged.
2. The two non-default constructors don't make sense to me. Both call mutators and then overwrite the data again afterwards. I assume that the mutators will already overwrite the data variables so you won't need to do it twice.

apainintheneck commented:
The merge conflicts are easy to fix. You can leave that to me once you fix the constructors.

brycerandolph committed:
Changed constructors

Changed two constructors to print when they return true or false and not
set values twice.

apainintheneck commented:
There are still problems with the constructors.

DataMatrix(BarcodeImage image)
-The scan() method only sets image, actualWidth, and actualHeight. You still need to set text to a default value.
-You can accomplish the test with one if statement.
if(scan(image) == false)
   //Set default values for image, actualWidth, and actualHeight.
-You don't need to include print statements in the constructor unless you are testing.

DataMatrix(String Text)
-The readText() method only sets text. You still need to set default values for image, actualWidth, and actualHeight.
-You can accomplish the test with one if statement.
if(readText(text) == false)
   //Set default values for text.
-You don't need to include print statements in the constructor unless you are testing.

Let me know if that makes sense.

brycerandolph committed:
Constructors fixed according to comments by @apainintheneck

brycerandolph committed:
removed extra "this."

apainintheneck committed:
Merge branch 'master' into Bryce-Phase3

apainintheneck commented:
Looks good to me. I'll merge it into into the master branch.
